File: dayOne/java/levelFour/TwoSumTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

public class TwoSumTest {

    @Test
    public void testThatGetTwoSumReturnsAnArrayOfTheIndexesThatEqualsTheSumOfTheTarget() {
        int[] numbers = {8, 4, -2, 7, 1};
        int target = 6;

        int[] expected = TwoSum.getTwoSum(numbers, target);
        int[] actual = {0, 2};

        assertArrayEquals(actual, expected);
    }

    @Test
    public void testThatGetTwoSumReturnsAnArrayOfTheFirstOccurrenceThatEqualsTheSumOfTheTarget() {
        int[] numbers = {2, 6, 2, -2, 9, 4, 7, 1};
        int target = 10;

        int[] expected = TwoSum.getTwoSum(numbers, target);
        int[] actual = {1, 5};

        assertArrayEquals(actual, expected);
    }
}

File: dayOne/python/levelFour/two_sum_test.py
from unittest import TestCase

from two_sum import get_two_sum

class TestTwoSum(TestCase):

    def test_that_get_two_sum_returns_an_array_of_index_of_numbers_that_equals_the_sum_of_the_target(self):

        numbers = [8, 4, -2, 7, 1]

        target = 6

        expected = get_two_sum(numbers, target)

        actual = [0, 2]

        self.assertListEqual(actual, expected)

    def test_that_get_two_sum_returns_an_array_of_index_of_numbers_of_the_first_occurrence_that_equals_the_sum_of_the_target(self):

        numbers = [2, 6, 2, -2, 9, 4, 7, 1]

        target = 10

        expected = get_two_sum(numbers, target)

        actual = [1, 5]

        self.assertListEqual(actual, expected)